src/scene_battle_rpg2k3.cpp

```diff
@@ -427,9 +428,15 @@ void Scene_Battle_Rpg2k3::UpdateAnimations() {
 					enemy_cursor->SetX(enemy->GetBattlePosition().x + sprite->GetWidth() / 2 + 2);
 					enemy_cursor->SetY(enemy->GetBattlePosition().y - enemy_cursor->GetHeight() / 2);
 
-					auto* state = enemy->GetSignificantState();
-					if (state) {
-						help_window->SetText(ToString(state->name), state->color);
+					std::vector<lcf::rpg::State*> ordered_states = enemy->GetInflictedStatesOrderedByPriority();
+					if (ordered_states.size() > 0) {
+						help_window->Clear();
+						int state_counter = 0;
+						for (lcf::rpg::State* state : ordered_states) {
+							std::string state_name = fmt::format("{:9s}", state->name);
+							help_window->AddText(state_name, state->color, Text::AlignLeft, false);
```

Comment:
What does RPG_RT do when thee state text runs off the edge of the window? Does it display partial text cutoff on the end like "poi" instead of "poison"? Or is it smart enough to not display the text if the entire word doesn't fit?

Reply:
Because RPG_RT shows at most 5 states (tested it myself), the state text will never run off the edge of the window. A state name can have at most 8 characters, and RPG_RT reserves 9 characters per state, so we have a maximum of 45 characters. The character limit of the help window is 50.